crates/tracing-instrumentation/src/prometheus_metrics.rs

```diff
@@ -74,7 +81,23 @@ impl Prometheus {
     }
 
     pub fn global_labels(&self) -> &Vec<String> {
-        &self.global_labels
+        if let Some(global_labels) = self.global_labels.get() {
+            return global_labels;
+        }
+
+        let Some(node_id) = GLOBAL_NODE_ID.get() else {
+            return &self.static_labels;
+        };
+
+        self.global_labels.get_or_init(|| {
+            let mut labels = self.static_labels.clone();
```

Comment:
Should we just take `static_labels`? `static_labels` should no longer be used afterwards, I assume.